### ISSUE TITLE
Style/ArgumentsForwarding mis-corrects when the called method has additional arguments.

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#12087]
Style/ArgumentsForwarding will replace all arguments to the called method with `...`, even if there are arguments besides the `*args, **kwargs, &`.

Given the following code:

```ruby
def self.get(*args, **kwargs, &)
  CanvasHttp.request(Net::HTTP::Get, *args, **kwargs, &)
end
```

## Expected behavior

```ruby
def self.get(...)
  CanvasHttp.request(Net::HTTP::Get, ...)
end
```

## Actual behavior

```ruby
def self.get(...)
  CanvasHttp.request(...)
end
```

Note the additional Net::HTTP::Get inline argument has been removed.

## Steps to reproduce the problem

Auto-correct the above code with the Style/ArgumentsForwarding cop.

## RuboCop version

```
1.55.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.3) [arm64-darwin21]
  - rubocop-factory_bot 2.23.1
  - rubocop-graphql 1.3.0
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.23.0
```

Note that I'm not sure if adding leading arguments is supported by all Ruby versions. I seem to remember when this syntax was introduced it was very limited. If that's the case, I would expect this cop to not even find an offense if the target Ruby version doesn't support leading arguments.

**Comment by owst** [comment]
> Note that I'm not sure if adding leading arguments is supported by all Ruby versions. I seem to remember when this syntax was introduced it was very limited. If that's the case, I would expect this cop to not even find an offense if the target Ruby version doesn't support leading arguments.

An excellent point - as per my testing, this was not supported in Ruby 2.7 until 2.7.3 [after it was backported from 3.0.0](https://bugs.ruby-lang.org/issues/16378). 

I think it'll be easiest to just not support forward-all-with-leading-args (and treat this as no offense) on Ruby < 3.0
